src/Raven.Client/Documents/Linq/RavenQueryProvider.cs

```diff
@@ -102,14 +104,15 @@ public IRavenQueryProvider For<TS>()
                 return this;
 
             var ravenQueryProvider = new RavenQueryProvider<TS>(
-                _queryGenerator,
-                _indexName,
-                _collectionName,
-                OriginalQueryType,
-                _queryStatistics,
-                _highlightings,
-                _isMapReduce,
-                _conventions);
+                    _queryGenerator,
+                    _indexName,
+                    _collectionName,
+                    OriginalQueryType,
+                    _queryStatistics,
+                    _highlightings,
+                    _isMapReduce,
+                    _conventions,
+                    FieldsToFetch?.Count == 0 ? null : FieldsToFetch);
```

Comment:
You can just the `FieldsToFetch` here, no?

Reply:
Yes, it is.
Done.